.github/workflows/ci.yml

```diff
@@ -166,79 +174,73 @@ jobs:
           arguments: --init-script gradle/support/fetchDependencies.gradle init
+
       # TODO: Pre-build GhidraDev
+
       - name: Checkout ghidra-data
         uses: actions/checkout@v4
         with:
           repository: NationalSecurityAgency/ghidra-data
-          path: 'ghidra-data'
-      - name: Copy ghidra-data files into the appropriate directories
+          path: ghidra-data
+
+      - name: Copy ghidra-data files
         run: cp -r ghidra-data/FunctionID/* Ghidra/Features/FunctionID/data
-      - name: Build ghidra, create a cross-platform distribution
-        uses: eskatos/gradle-command-action@v2
+
+      - name: Build Ghidra
+        uses: gradle/gradle-build-action@v2
         with:
-          gradle-version: 7.3.3
+          gradle-version: ${{ env.GRADLE_VERSION }}
           arguments: -x ip -PallPlatforms buildGhidra
+
       # TODO: remove upload-artifact when create release is sure to work
-      - name: Upload final dist
-        uses: actions/upload-artifact@v3
-        with:
-          path: "build/dist/*"
-      - name: Remove temporary artifacts
-        uses: geekyeggo/delete-artifact@v2
-        with:
-          name: |
-            natives-win_x86_32
-            natives-win_x86_64
-            natives-mac_x86_64
-            natives-mac_arm_64
-            natives-linux_x86_64
-            natives-linux_arm_64
+      - name: Upload distribution
+        uses: actions/upload-artifact@v4
+        with:
+          path: build/dist/*
+
       - name: Get current date, rev and dist name
         id: date
         run: |
           echo date=$(date +'%Y-%m-%d') >> $GITHUB_OUTPUT
           echo dist=$(ls build/dist) >> $GITHUB_OUTPUT
           echo rev=$(git rev-parse --short HEAD) >> $GITHUB_OUTPUT
-      - uses: actions/checkout@v4
+
+      - name: Checkout ghidra-ci
+        uses: actions/checkout@v4
         with:
-          path: ghidra-ci-roblabla
+          path: ghidra-ci
+
       - name: Generate CHANGELOG.md
+        if: ${{ inputs.prevRev != '' }}
         run: |
-          cd ghidra-ci-roblabla
+          cd ghidra-ci
           sudo apt-get update
           sudo apt-get install libkrb5-dev
           npm i
-          node generate_changelog.js ${{github.event.inputs.prevRev}} ${{github.event.inputs.rev}} > CHANGELOG.md
+          node generate_changelog.js ${{ inputs.repo }} ${{ inputs.prevRev }} ${{ inputs.rev }} > CHANGELOG.md
+
+      - name: Generate fallback CHANGELOG.md
+        if: ${{ inputs.prevRev == '' }}
+        run: |
+          cd ghidra-ci
+          echo "# Changelog" > CHANGELOG.md
+          echo "Built from [${{ inputs.repo }}@${{ inputs.rev }}](https://github.com/${{ inputs.repo }}/commit/${{ inputs.rev }})" >> CHANGELOG.md
+
       - name: Create Release
-        id: create_release
-        uses: actions/create-release@v1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        uses: softprops/action-gh-release@v1
```

Comment:
I'd rather keep using the deprecated `actions/create-release` instead of depending on a new, unknown author.

Reply:
It's 3 years out of date and makes the build throw warnings, which doesn't seem ideal. I wonder what the security boundary is here, can an action access secrets that you haven't passed to it? (e.g. Apple creds)

Reply:
I notice this step (Ghidra dist) doesn't deal with anything sensitive either, the notarization/signing is in the previous step. Happy to revert if it's still undesired